lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -1578,6 +1584,20 @@ maybe_warn_for_ambiguous_bang_before_equals(_Kind, _Atom, _Rest, _Line, _Column,
 prepend_warning(Line, Column, File, Msg, #elixir_tokenizer{warnings=Warnings} = Scope) ->
   Scope#elixir_tokenizer{warnings = [{{Line, Column}, File, Msg} | Warnings]}.
 
+track_ascii(Ascii, #elixir_tokenizer{ascii_identifiers_only=AsciiOnly} = Scope) ->
+  Scope#elixir_tokenizer{ascii_identifiers_only=Ascii and AsciiOnly}.
+
+maybe_warn_on_unicode_security(Tokens, File, #elixir_tokenizer{ascii_identifiers_only=false, identifier_tokenizer=IdentifierTokenizer, warnings=Warnings} = Scope) ->
+  UnicodeWarnings =
+    case erlang:function_exported(IdentifierTokenizer, unicode_lint_warnings, 1) of
```

Comment:
In fact, I would make it call something like `'String.Confusable':lint(...)` and decouple from the tokenizer. :)

Reply:
> it will never be called by the bare tokenizer since it only handles ascii anyway. :)

Oh yeah, can avoid that check now -- before I was calling it at end of every file, heh. Re: naming, and splitting it out of tokenizer, makes sense to me; I have a comment below asking about the right way to do that. Re: String as top namespace, actually the script resolution IS something that I'd love to have available in String eventually, even if right now it'll be an undocumented impl. detail like Tokenizer.tokenize. Rust's unicode-scripts provides that for all strings, and now some stuff like username validation, I want my validation rule to be 'single-script usernames only', following uts39's definition of ss.